gst/nnstreamer/tensor_source/tensor_src_iio.c

```diff
@@ -1225,17 +1225,24 @@ gst_tensor_src_iio_set_property (GObject * object, guint prop_id,
           val = g_ascii_strtoll (strv[i], &endptr, 10);
           if (errno == ERANGE || errno == EINVAL || (endptr == strv[i]
                   && val == 0)) {
-            GST_ERROR_OBJECT (self, "Cannot parse received custom channels %s",
+            GST_ERROR_OBJECT (self,
+                "Cannot parse received custom channels %s. The property values for CHANNELS are ignored.",
                 param);
             g_hash_table_destroy (self->custom_channel_table);
             self->custom_channel_table = NULL;
+            status = FALSE;
             break;
           }
-          status = g_hash_table_insert (self->custom_channel_table,
-              GINT_TO_POINTER (val), NULL);
-          g_assert (status);
+          if (FALSE == g_hash_table_insert (self->custom_channel_table,
+                  GINT_TO_POINTER (val), NULL)) {
+            /** this means val is duplicated. just skip it, then. */
+            ml_logw
+                ("tensor-src-iio's CHANNELS property balue has duplicated entry, '%s', which is ignored.\n",
```

Comment:
g_hash_table_insert replaces it's previous value in case there is a duplicated entry. So it is not ignored.

https://developer.gnome.org/glib/stable/glib-Hash-Tables.html

Reply:
I think there is a typo: `balue` -> `value`

Reply:
- typo fixed.
- It's an integer value. Thus, I rephrased: ignored --> registered only once.

Reply:
Thanks for the explanation :)